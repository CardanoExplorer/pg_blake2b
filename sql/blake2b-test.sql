--
-- BLAKE2B family
--
CREATE EXTENSION blake2b;

-- BLAKE2B_224
SELECT encode(blake2b('', 28), 'hex');
SELECT encode(blake2b(''::bytea, 28), 'hex');
SELECT encode(blake2b('\x'::bytea, 28), 'hex');
SELECT encode(blake2b('\x', 28), 'hex');
SELECT encode(blake2b('33', 28), 'hex');
SELECT encode(blake2b('\x3333'::bytea, 28), 'hex');
SELECT encode(blake2b('3333', 28), 'hex');
SELECT encode(blake2b('28 bytes digest with key', 28, '\x0123456789'::bytea), 'hex');

-- BLAKE2B_256
SELECT encode(blake2b('', 32), 'hex');
SELECT encode(blake2b(''::bytea, 32), 'hex');
SELECT encode(blake2b('\x'::bytea, 32), 'hex');
SELECT encode(blake2b('\x', 32), 'hex');
SELECT encode(blake2b('33', 32), 'hex');
SELECT encode(blake2b('\x3333'::bytea, 32), 'hex');
SELECT encode(blake2b('3333', 32), 'hex');
SELECT encode(blake2b('32 bytes digest with key', 32, '\x0123456789'::bytea), 'hex');

-- BLAKE2B_384
SELECT encode(blake2b('', 48), 'hex');
SELECT encode(blake2b(''::bytea, 48), 'hex');
SELECT encode(blake2b('\x'::bytea, 48), 'hex');
SELECT encode(blake2b('\x', 48), 'hex');
SELECT encode(blake2b('33', 48), 'hex');
SELECT encode(blake2b('\x3333'::bytea, 48), 'hex');
SELECT encode(blake2b('3333', 48), 'hex');
SELECT encode(blake2b('48 bytes digest with key', 48, '\x0123456789'::bytea), 'hex');

-- BLAKE2B_512
SELECT encode(blake2b(''), 'hex');
SELECT encode(blake2b('', 64), 'hex');
SELECT encode(blake2b(''::bytea, 64), 'hex');
SELECT encode(blake2b('\x'::bytea, 64), 'hex');
SELECT encode(blake2b('\x', 64), 'hex');
SELECT encode(blake2b('33', 64), 'hex');
SELECT encode(blake2b('\x3333'::bytea, 64), 'hex');
SELECT encode(blake2b('3333', 64), 'hex');
SELECT encode(blake2b('64 bytes digest with key', 64, '\x0123456789'::bytea), 'hex');
