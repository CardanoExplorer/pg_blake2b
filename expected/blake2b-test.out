--
-- BLAKE2B family
--
CREATE EXTENSION blake2b;
-- BLAKE2B_224
SELECT encode(blake2b('', 28), 'hex');
                          encode                          
----------------------------------------------------------
 836cc68931c2e4e3e838602eca1902591d216837bafddfe6f0c8cb07
(1 row)

SELECT encode(blake2b(''::bytea, 28), 'hex');
                          encode                          
----------------------------------------------------------
 836cc68931c2e4e3e838602eca1902591d216837bafddfe6f0c8cb07
(1 row)

SELECT encode(blake2b('\x'::bytea, 28), 'hex');
                          encode                          
----------------------------------------------------------
 836cc68931c2e4e3e838602eca1902591d216837bafddfe6f0c8cb07
(1 row)

SELECT encode(blake2b('\x', 28), 'hex');
                          encode                          
----------------------------------------------------------
 c8515c42ae0b19f21f7c70bf89e63a457466c5b330f119aac8b266aa
(1 row)

SELECT encode(blake2b('33', 28), 'hex');
                          encode                          
----------------------------------------------------------
 bc5d994c5dbdc25a0f62cdd0e8fe2f0ec8665afd743a0cb364a807ff
(1 row)

SELECT encode(blake2b('\x3333'::bytea, 28), 'hex');
                          encode                          
----------------------------------------------------------
 bc5d994c5dbdc25a0f62cdd0e8fe2f0ec8665afd743a0cb364a807ff
(1 row)

SELECT encode(blake2b('3333', 28), 'hex');
                          encode                          
----------------------------------------------------------
 30f57835bf541569ad0403a213fdb4d33ac2788a97d48e6885f58952
(1 row)

SELECT encode(blake2b('28 bytes digest with key', 28, '\x0123456789'::bytea), 'hex');
                          encode                          
----------------------------------------------------------
 d31636a38b7539138e6a76c9179ce36b6d9d49b5bf967c76e8cbd0d3
(1 row)

-- BLAKE2B_256
SELECT encode(blake2b('', 32), 'hex');
                              encode                              
------------------------------------------------------------------
 0e5751c026e543b2e8ab2eb06099daa1d1e5df47778f7787faab45cdf12fe3a8
(1 row)

SELECT encode(blake2b(''::bytea, 32), 'hex');
                              encode                              
------------------------------------------------------------------
 0e5751c026e543b2e8ab2eb06099daa1d1e5df47778f7787faab45cdf12fe3a8
(1 row)

SELECT encode(blake2b('\x'::bytea, 32), 'hex');
                              encode                              
------------------------------------------------------------------
 0e5751c026e543b2e8ab2eb06099daa1d1e5df47778f7787faab45cdf12fe3a8
(1 row)

SELECT encode(blake2b('\x', 32), 'hex');
                              encode                              
------------------------------------------------------------------
 8542024791610a1bfcecf6985b0bbf48a2b6753e080c2ec8a316b6c757b59560
(1 row)

SELECT encode(blake2b('33', 32), 'hex');
                              encode                              
------------------------------------------------------------------
 ef97c9e36f5e33370c795599501e8f834363461eb001fd227831992c57b5bedf
(1 row)

SELECT encode(blake2b('\x3333'::bytea, 32), 'hex');
                              encode                              
------------------------------------------------------------------
 ef97c9e36f5e33370c795599501e8f834363461eb001fd227831992c57b5bedf
(1 row)

SELECT encode(blake2b('3333', 32), 'hex');
                              encode                              
------------------------------------------------------------------
 f4395a05f95dbb8a628c7be3eb5cbcde9393d90fd44837ab390c027daecc6c33
(1 row)

SELECT encode(blake2b('32 bytes digest with key', 32, '\x0123456789'::bytea), 'hex');
                              encode                              
------------------------------------------------------------------
 912bf54103cb9d1093c3e736514040fd6a7ea75d094c8ce843425a4791571b82
(1 row)

-- BLAKE2B_384
SELECT encode(blake2b('', 48), 'hex');
                                              encode                                              
--------------------------------------------------------------------------------------------------
 b32811423377f52d7862286ee1a72ee540524380fda1724a6f25d7978c6fd3244a6caf0498812673c5e05ef583825100
(1 row)

SELECT encode(blake2b(''::bytea, 48), 'hex');
                                              encode                                              
--------------------------------------------------------------------------------------------------
 b32811423377f52d7862286ee1a72ee540524380fda1724a6f25d7978c6fd3244a6caf0498812673c5e05ef583825100
(1 row)

SELECT encode(blake2b('\x'::bytea, 48), 'hex');
                                              encode                                              
--------------------------------------------------------------------------------------------------
 b32811423377f52d7862286ee1a72ee540524380fda1724a6f25d7978c6fd3244a6caf0498812673c5e05ef583825100
(1 row)

SELECT encode(blake2b('\x', 48), 'hex');
                                              encode                                              
--------------------------------------------------------------------------------------------------
 04815bce48b7a8660efd88e8cc19bce2b3c266fd18bd1f77a041cc2f1f73c1f58807d551d6b2c10f764cab8dc874be79
(1 row)

SELECT encode(blake2b('33', 48), 'hex');
                                              encode                                              
--------------------------------------------------------------------------------------------------
 5a14b4f2f75ad90dadb5f500ec0813938460e2ed63306e2bbcd6381c4dde16f1a9c28fa8a74733cae364a62dd8cbb073
(1 row)

SELECT encode(blake2b('\x3333'::bytea, 48), 'hex');
                                              encode                                              
--------------------------------------------------------------------------------------------------
 5a14b4f2f75ad90dadb5f500ec0813938460e2ed63306e2bbcd6381c4dde16f1a9c28fa8a74733cae364a62dd8cbb073
(1 row)

SELECT encode(blake2b('3333', 48), 'hex');
                                              encode                                              
--------------------------------------------------------------------------------------------------
 1771d47b49740c888af28f36274f136e11ef9d015e80e227332bbcbee33f21711d6aa0fa9e2f98dcdf584fd5575c54ff
(1 row)

SELECT encode(blake2b('48 bytes digest with key', 48, '\x0123456789'::bytea), 'hex');
                                              encode                                              
--------------------------------------------------------------------------------------------------
 01d0e42e439bfd69bb914815ccd74afbca19520b6364a882671d2414b1636ad52231b5d8ba8d77e189369bb4895792e6
(1 row)

-- BLAKE2B_512
SELECT encode(blake2b(''), 'hex');
                                                              encode                                                              
----------------------------------------------------------------------------------------------------------------------------------
 786a02f742015903c6c6fd852552d272912f4740e15847618a86e217f71f5419d25e1031afee585313896444934eb04b903a685b1448b755d56f701afe9be2ce
(1 row)

SELECT encode(blake2b('', 64), 'hex');
                                                              encode                                                              
----------------------------------------------------------------------------------------------------------------------------------
 786a02f742015903c6c6fd852552d272912f4740e15847618a86e217f71f5419d25e1031afee585313896444934eb04b903a685b1448b755d56f701afe9be2ce
(1 row)

SELECT encode(blake2b(''::bytea, 64), 'hex');
                                                              encode                                                              
----------------------------------------------------------------------------------------------------------------------------------
 786a02f742015903c6c6fd852552d272912f4740e15847618a86e217f71f5419d25e1031afee585313896444934eb04b903a685b1448b755d56f701afe9be2ce
(1 row)

SELECT encode(blake2b('\x'::bytea, 64), 'hex');
                                                              encode                                                              
----------------------------------------------------------------------------------------------------------------------------------
 786a02f742015903c6c6fd852552d272912f4740e15847618a86e217f71f5419d25e1031afee585313896444934eb04b903a685b1448b755d56f701afe9be2ce
(1 row)

SELECT encode(blake2b('\x', 64), 'hex');
                                                              encode                                                              
----------------------------------------------------------------------------------------------------------------------------------
 bd1f9da8bca03d0c9c78d69269195c108bd80a2bbd6dbeba845c4c4dcd105586de986c5437f50d07ad8ffaa4b42108220c4aab15ea10e0730c5437e033d1976d
(1 row)

SELECT encode(blake2b('33', 64), 'hex');
                                                              encode                                                              
----------------------------------------------------------------------------------------------------------------------------------
 afd85ebe3a733ccf64c11c681e05595490854c4a485be585dfff1f2c1958823a0fa11069467c82602b3fece173df052206012b8df3a90b68569c32b3b85c817c
(1 row)

SELECT encode(blake2b('\x3333'::bytea, 64), 'hex');
                                                              encode                                                              
----------------------------------------------------------------------------------------------------------------------------------
 afd85ebe3a733ccf64c11c681e05595490854c4a485be585dfff1f2c1958823a0fa11069467c82602b3fece173df052206012b8df3a90b68569c32b3b85c817c
(1 row)

SELECT encode(blake2b('3333', 64), 'hex');
                                                              encode                                                              
----------------------------------------------------------------------------------------------------------------------------------
 467caae5ccc9e93d82f5c6c7fdbac4bfda6ee429fe68131fdab17091e6c1878dc31835b3bcb4296ea457d47f37be6c4ecd14dadfa22cf28e271ab07380a3deff
(1 row)

SELECT encode(blake2b('64 bytes digest with key', 64, '\x0123456789'::bytea), 'hex');
                                                              encode                                                              
----------------------------------------------------------------------------------------------------------------------------------
 342d640078aefbd1dcaae54a020918faea424c0ced682a634f3b59f6c366071a2a2a693b91e7bfc5997da69dbf23921c0aba9313423b1df40bfaf7fceb3149dd
(1 row)

